cmd/generate/config/rules/kubernetes.go

```diff
@@ -0,0 +1,62 @@
+package rules
+
+import (
+	"regexp"
+
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+// The kubernetes rules are split into two functions to make the complex proximity matching of the data-key and the kind-identifier more readable and testable
+
+// KubernetesSecretWithDataBefore validates if we detected a kubernetes secret which contains data, before the resource identifier!
+func KubernetesSecretWithDataBefore() *config.Rule {
+	// define rule
+	r := config.Rule{
+		RuleID:      "kubernetes-secret-with-data-before",
+		Description: "Possible Kubernetes Secret detected, posing a risk of leaking credentials/tokens from your deployments",
+		// We try to match secrets by looking if we have the keyword
+		Regex: generateUniqueTokenRegex(`(?i)(?:\b(data)\b)\W+(?:\w+\W+){0,200}?\bkind:\s*Secret\b.*`, true),
```

Comment:
I _think_ this could be improved to explicitly match `data:` instead of just `data` -- unless you have an example of why that's a bad idea.

You could also include the content between `data:` and `kind` in the capture, rather than just data. Rough example: https://regex101.com/r/2Qu5ho/1

Reply:
Implemented the suggestion with minor changes